projects/avodata/AvodataSolverTool.java

```diff
@@ -0,0 +1,142 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Scanner;
+
+import org.apache.commons.math3.analysis.function.Abs;
+import org.apache.commons.math3.linear.Array2DRowRealMatrix;
+import org.apache.commons.math3.linear.DecompositionSolver;
+import org.apache.commons.math3.linear.MatrixUtils;
+import org.apache.commons.math3.linear.RealMatrix;
+import org.apache.commons.math3.linear.RealVector;
+import org.apache.commons.math3.linear.SingularValueDecomposition;
+import org.apache.commons.math3.util.CombinatoricsUtils;
+
+/** To build: bazel build //projects/avodata:ls-alg_tool To run: bazel-bin/projects/avodata/ls-alg_tool */
+/** This class reads matrix from csv file, ignores the parmeters and calculate 
+/** using least squares */
+public class AvodataSolverTool {
+
+  static final String fpath = "/projects/avodata/check.csv";
+
+  public static double factorial(int n) {
+    return CombinatoricsUtils.factorialDouble(n);
+  }
+
+  public static double nChooseK(int n, int k) {
+    return CombinatoricsUtils.binomialCoefficientDouble(n, k);
+  }
+
+  public static double[][] generateSyntheticA(int numOfProfessions) {
+
+    int numOfraws = (int) nChooseK(numOfProfessions, 2);
+    RealMatrix matrix = new Array2DRowRealMatrix(numOfraws, numOfProfessions);
+
+    int row = 0;
+    for (int i = 0; i < numOfProfessions; i++)
+      for (int j = i + 1; j < numOfProfessions; j++) {
+        matrix.addToEntry(row, i, 1);
+        matrix.addToEntry(row, j, -1);
+        row++;
+      }
+    return matrix.getData();
+  }
+
+  public static double[][] generateSyntethicC(RealMatrix m, int i, RealMatrix a) {
+    RealMatrix c = a.multiply(m.getRowMatrix(i).transpose());
+    return c.getData();
+  }
+
+  public static RealVector solve(RealMatrix a, RealVector c) {
+
+    SingularValueDecomposition s = new SingularValueDecomposition(a);
+    DecompositionSolver d = s.getSolver();
+    return d.solve(c);
+  }
+
+  public static double[][] buildCheck(
+      int numOfProfessions, RealMatrix cMatrix, int numOfCharacteristics) {
+    RealMatrix a = new Array2DRowRealMatrix();
+    a = MatrixUtils.createRealMatrix(generateSyntheticA(numOfProfessions));
+    RealMatrix m = new Array2DRowRealMatrix(numOfCharacteristics, numOfProfessions);
+
+    for (int i = 0; i < numOfCharacteristics; i++) {
+
+      RealVector c = cMatrix.getRowVector(i);
+      RealVector rowInM = solve(a, c);
+      Abs av = new Abs();
```

Comment:
Please use descriptive names for variables, unless it's super-common (like `i` for loops).
In this case, I can guess this means `absolute value`, but I'm not sure, and it's easier to read when it's a full word or words.